core/modules/simpletest/tests/common.test

```diff
@@ -1390,6 +1390,129 @@ class CommonJavaScriptTestCase extends BackdropWebTestCase {
     $this->assertTrue(strpos($javascript, $inline) > 0, 'Rendered JavaScript footer returns the inline code.');
   }
 
+  /**
+   * Test the 'javascript_always_use_jquery' setting.
+   */
+  function testJavaScriptAlwaysUseJQuery() {
+    // The default front page of the site should use jQuery and other standard
+    // scripts and settings.
+    $this->backdropGet('');
+    $this->assertRaw('core/misc/jquery.js', 'Default behavior: The front page of the site includes jquery.js.');
+    $this->assertRaw('core/misc/backdrop.js', 'Default behavior: The front page of the site includes backdrop.js.');
+    $this->assertRaw('window.Backdrop', 'Default behavior: The front page of the site includes Backdrop settings.');
+    $this->assertRaw('basePath', 'Default behavior: The front page of the site includes the basePath Backdrop setting.');
+
+    // The default front page should not use jQuery and other standard scripts
+    // and settings when the 'javascript_always_use_jquery' setting is set to
+    // FALSE.
+    state_set('javascript_always_use_jquery', FALSE);
```

Comment:
@klonos this needs to be set in a different way. The code uses `settings_get` so you can't use `state_set` here. `settings_get` is probably the right choice but I don't know how to set it in a test. Maybe search core for examples.

Reply:
Thanks for reviewing @herbdool 🙏 ...I understand that this is not the right way. I have used `state_set()` and `state_del()` temporarily here, until I figured out how to dynamically set something that would normally live in settings.php, during a test run.

It seems to me that currently there is no "delicate" way to do this (pending https://github.com/backdrop/backdrop-issues/issues/2805 or any solution that will allow overriding things set in settings.php), so I might resort to adding a fallback to `state_get()` in the conditional that checks for `javascript_always_use_jquery` in `backdrop_add_js()`. That way, I will be able to use `state_set()`/`state_del()` in the test here.

Reply:
You're welcome. The correct way of doing this now is to use both `settings_get` and then `config_get`. `settings_get` should always override the latter. That's how it worked in D7. You can see an example here https://github.com/backdrop-contrib/reroute_email/blob/1.x-1.x/reroute_email.admin.inc (though it doesn't need to be as complex).

Reply:
Also, you don't need to use `state_*` at all since you can get the result we want by other means.